frontend/util/list-search/search-dsl.helpers.tsx

```diff
@@ -1,13 +1,13 @@
 import queryString from "query-string";
 
-export const allowedSearchFields = {
-  id: "Client ID",
-  zip: "ZIP Code",
-  phone: "Phone",
-  program: "Interest in Program"
-};
+export let allowedSearchFields = {};
+
+let allowedKeys = Object.keys(allowedSearchFields);
 
-const allowedKeys = Object.keys(allowedSearchFields);
+export function setAllowedSearchFields(searchFields) {
```

Comment:
This has a bug: it does not allow for two react components to simultaneously use the search dsl helpers. Their react effects would call setAllowedSearchFields at different times, but not everytime before calling `parseSearch`, which would mean that sometimes the wrong searchFields would apply.

Instead, let's add the searchFields as an argument to the `parseSearch`, `serializeSearch`, and `deserializeSearch` functions.

The alternative would be something like this, which I think is a more substantial refactor than is worth it:

```js
// each call to dslSearcher returns a searcher that is bound to those allowedSearchFields.
const dslSearcher = new DslSearcher(allowedSearchFields)
dslSearcher.parseSearch()
```